pkg/api/vm/handler.go

```diff
@@ -593,6 +617,81 @@ func (h *vmActionHandler) updateVMVolumeClaimTemplate(vm *kubevirtv1.VirtualMach
 	return nil
 }
 
+func (h *vmActionHandler) getPVCStorageClassMap(vm *kubevirtv1.VirtualMachine) (map[string]string, error) {
```

Comment:
Question: Why don't we just get the pvc's storageclass from `pvc.Spec.StorageClassName`

Reply:
If a PVC is from a VMImage, its `pvc.Spec.StorageClassName` is the storage class of a VMImage.

Reply:
If this pvc is not from the image, should we get it from pvc.Spec.StorageClassName? not default storageclass

Reply:
This default storageclass setting should be deprecated

Reply:
Yes, we can get `pvc.Spec.StorageClassName` directly when a PVC is not from an image. However, if a PVC is from an image, we may still need the default storageclass setting, because we can't guarantee `harvesterhci.io/storageClassName` annotation is in VMImage.

Reply:
we should get the default storageclass from the storageclass list
```go
	storageClasses, err := h.storageClassCache.List(labels.Everything())
	if err != nil {
		return nil, err
	}

	for _, storageClass := range storageClasses {
		if storageClass.Annotations[util.AnnotationIsDefaultStorageClassName] == "true" {

		}
	}
```

Reply:
Thanks for your information. I didn't know that we use `AnnotationIsDefaultStorageClassName` in VMImage. I will update it!